src/development/platform-integration/ios/app-extensions.md

```diff
@@ -0,0 +1,213 @@
+the `Runner` app target and the extension target in the same
+[App Group][].
+
+To add a target to an App Group:
+
+1. Open the target settings in Xcode.
+1. Navigate to the **Signing & Capabilities** tab.
+1. Select **+ Capability** then **App Groups**.
+1. Select or create the App Group.
+
+{{site.alert.note}}
+  You must be signed in to your Apple Developer account.
+{{site.alert.end}}
+
+{% include docs/app-figure.md
+image="development/platform-integration/app-extensions/xcode-app-groups.png" %}
+
+When two targets belong to the same App Group, they can read and write
+data to the same container using one of the following
+options:
+
+- **Key/value:** Use the [`shared_preference_app_group`][]
+  plugin to read or write to `UserDefaults` within the same App Group.
+- **File:** Use the App Group container path from the
+  [`path_provider`][] plugin to [read and write files][].
+- **Database:** Use the App Group container path from
+  the [`path_provider`][] plugin to create a database with the
+  [`sqflite`][] plugin.
+
+### Background updates
+Background tasks provide a means to update your extension through code
+regardless of the status of your app.
+
+To schedule background work from your Flutter app, use the
+[`workmanager`][] plugin.
+
+### Deep linking
+You might want to direct users from an app extension to a
+specific page in your Flutter app.
+To have a URL open a specified route in your app, you can use
+[Deep Linking][].
+
+## Creating app extension UIs with Flutter
+Some app extensions display a user interface.
+For example, iMessage extensions allow users to access your app's
+content directly from the **Messages** app.
+
+<figure class="site-figure {{include.class}}">
+    <div class="site-figure-container">
+        <img src='/assets/images/docs/development/platform-integration/app-extensions/imessage-extension.png'
+        height='300'>
+    </div>
+</figure>
+
+The Flutter command line tool does **not** support 
+building Flutter UI for app extensions. 
+To create the UI for 
+an app extension using Flutter, you must compile 
+a custom engine and embed the `FlutterViewController`
+as described in the following section.
+
+{{site.alert.note}}
+  This requires a custom build of the Flutter engine and
+  is only for advanced users.
+  To learn more, check out [Compiling the engine][].
+{{site.alert.end}}
+
+<!-- TO DO: replace link with a ghost PR -->
+1. Create a custom build of the Flutter engine that removes uses of
+   `sharedApplication` and corrects the path for the bundle.
+   Check out an [example from the community on GitHub][].
+
+1. Open the Flutter app project settings in Xcode to share build
+   configurations. 
+
+   In the **Info** tab,
+   under the **Configurations** expandable group,
+   expand the **Debug**, **Profile**, and **Release** entries.
+   For each, select the same value from the drop-down menu for the
+   extension target as the entry selected for the normal app target. 
```

Comment:
**issue:** This reads with difficulty. I'm unsure -- without the image -- what you need me to do exactly.
```suggestion
   1. Navigate to the **Info** tab.
   1. Expand the **Configurations** group. 
   1. Expand the **Debug**, **Profile**, and **Release** entries.
   1. For each of these configurations, make sure the value in the
       **Based on configuration file** drop-down menu for your
       extension matches the one selected for the normal app target.
```